miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/SampleRestController.java

```diff
@@ -324,17 +297,15 @@ public SampleDto updateSample(@PathVariable("id") Long id, @RequestBody SampleDt
     return identitiesBySearchTerm;
   }
 
-  private Set<SampleDto> getSamplesForIdentityString(String identityIdentifier, String project, boolean permitPartialMatch)
+  private Set<SampleDto> getSamplesForIdentityString(String identityIdentifier, String project, boolean exactMatch)
       throws IOException {
     Collection<SampleIdentity> matches = new HashSet<>();
     Project selected = null;
-    if (!LimsUtils.isStringEmptyOrNull(project)) selected = projectService.getProjectByShortName(project);
+    selected = projectService.getProjectByShortName(project);
     if (selected != null) {
-      matches = sampleService.getIdentitiesByExactExternalNameAndProject(identityIdentifier, selected.getId());
-    } else if (permitPartialMatch) {
-      matches = sampleService.getIdentitiesByExternalNameOrAlias(identityIdentifier);
+      matches = sampleService.getIdentitiesByExternalNameOrAliasAndProject(identityIdentifier, selected.getId(), exactMatch);
     } else {
-      matches = sampleService.getIdentitiesByExactExternalName(identityIdentifier);
+      matches = sampleService.getIdentitiesByExternalNameOrAliasAndProject(identityIdentifier, null, exactMatch);
     }
     return matches.stream().map(identity -> Dtos.asDto(identity)).collect(Collectors.toSet());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)